src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.cs

```diff
@@ -178,8 +180,22 @@ private void InitializeTemplate()
 
 			if (GetTemplateChild(c_numberBoxTextBoxName) is TextBox textBox)
 			{
-				textBox.KeyDown += OnNumberBoxKeyDown;
-				registrations.Add(() => textBox.KeyDown -= OnNumberBoxKeyDown);
+				if (SharedHelpers.IsRS3OrHigher())
+				{
+					// Listen to PreviewKeyDown because textbox eats the down arrow key in some circumstances.
+					// UNO Docs: PreviewKeyDown is not implemented. Use KeyDown.
+					//textBox.PreviewKeyDown += OnNumberBoxKeyDown;
+					//registrations.Add(() => textBox.PreviewKeyDown -= OnNumberBoxKeyDown);
+					textBox.KeyDown += OnNumberBoxKeyDown;
+					registrations.Add(() => textBox.KeyDown -= OnNumberBoxKeyDown);
```

Comment:
Not sure this is okay or the correct way to do it.

Reply:
This is is the same block as below, let's put this block around `#if !HAS_UNO` for now

Reply:
@jeromelaban Does the change in 11d2266 match what you meant?

Reply:
That's correct! Thanks!